apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ast/builder/SchemaBuilder.kt

```diff
@@ -19,16 +18,26 @@ internal fun CodeGenerationIR.ast(
     useSemanticNaming: Boolean,
     operationOutput: OperationOutput
 ): Schema {
-  val enums = typesUsed.filter { it.kind == TypeDeclaration.KIND_ENUM }.map { it.ast() }
-  val inputTypes = typesUsed.filter { it.kind == TypeDeclaration.KIND_INPUT_OBJECT_TYPE }.map {
+  val enums = typeDeclarations.filter {
+    it.kind == TypeDeclaration.KIND_ENUM
+        // && enumsToGenerate.contains(it.name) // filtering is done later as the rest of the codegen needs all enums
+  }.map {
+    it.ast()
+  }
+  val inputTypes = typeDeclarations.filter {
+    it.kind == TypeDeclaration.KIND_INPUT_OBJECT_TYPE
+        && inputObjectsToGenerate.contains(it.name)
+  }.map {
     it.ast(
         enums = enums,
         customTypeMap = customTypeMap,
         typesPackageName = typesPackageName
     )
   }
   val irFragments = fragments.associateBy { it.fragmentName }
-  val fragments = fragments.map {
+  val fragments = fragments.filter {
+    fragmentsToGenerate.contains(it.fragmentName)
```

Comment:
same for fragments (all fragments are sill needed as input to validate arguments, type conditions, etc...)